lib/package.json

```diff
@@ -47,8 +47,9 @@
     "@apollo/utils.createhash": "^3.0.0",
     "mammoth": "^1.6.0",
     "p-queue": "^7.4.1",
+    "pdfjs-dist": "^4.2.67",
```

Comment:
You _probably_ don't need to add a dependency to pdf.js, as it's already bundled in Obsidian and you should be able to use it with something like this.

```ts
const arrayBuffer = await app.vault.readBinary(file);
// @ts-ignore
const document = await window.pdfjsLib.getDocument(arrayBuffer).promise;
for (let i = 1; i <= document.numPages; i++) {
  const page = await document.getPage(i);
  // etc.
```

**But!** The bundled version might cause issues so it's still worth trying with an external dependency 👍

Reply:
i'll try the bundled version first